### ISSUE TITLE
Ideas for updating the website

### ISSUE THREAD

**Issue by thomasaarholt** [hyperspy/hyperspy#1449]
It would be nice to give the website a facelift, if only to get it functioning well on mobile. Does anyone have any thoughts on this? 

We currently use Sphinx to generate the website from reStructuredText (.rst) documents, with the built-in *agogo* sphinx theme (css).

Personally I would prefer it if we were working off the markdown syntax rather than .rst, but it really doesn't matter much.

**Comment by thomasaarholt** [comment]
I've had a play with changing the theme for the sphinx to one of the bootstrap ones, see https://thomasaarholt.github.io/. It looks like there is quite a bit missing still to get it mobile-friendly. Also, the navbar doesn't update correctly, making me question how it was put together in the first place.

**Comment by francisco-dlp** [comment]
Personally I am all for revamping the website using e.g. https://github.com/jekyll/jekyll . These days people are used to parse [websites like this one](https://wallabag.org/en) rather than our XXth century looking one...

**Comment by thomasaarholt** [comment]
Okay. I'm gonna let the creative juices flow and see if I can wield my newfound thesis-free freedom in the direction of Jekyll.

**Comment by magnunor** [comment]
A more minimalist approach can be done using readthedocs theme, for example: http://atomap.org/

However, I agree site like https://wallabag.org/en look really nice.

**Comment by francisco-dlp** [comment]
:)

Hold on until we reach some consensus. I am not very up-to-date with web design so my jekyll suggestions may be off-track. I just thought that it must be a good idea because it works seamlessly with github pages and it seems to be simple.

**Comment by thomasaarholt** [comment]
No worries. I'm just enjoying seeing what web development uses nowadays. Wallabag specifically seems to use [Pelican](http://docs.getpelican.com/en/stable/), btw.

**Comment by to266** [comment]
...Actually I do not like wallabag website myself - too many clicks / scrolling (they are equivalent!! When you click on a link, it actually just scrolls (__painfully slowly__) to the correct position, where you have to click again) is required to find anything, and things initially look just like a picture, not a source of information*.

While the idea is very similar, I believe [jupyter](http://jupyter.org/) did a way better job - while it stills has the same "scrolly" feel, there are obvious yet non-intrusive buttons at the top that work as expected (i.e. not scrolling) as far as I can tell.

*The one thing I really like about wallabag, though, is that the main idea is clearly stated and obvious once the website is open, whereas jupyter just promotes its conference. That may be just the play of the size - if you're small, you have to let people know what this is that they found. If you're big, you can forgo such "pleasantries", and just push whatever you think is the most relevant thing at the time.

**Comment by francisco-dlp** [comment]
I fully agree with @to266 insightful comment. By the way, jupyter uses jekyll https://github.com/jupyter/jupyter.github.io

**Comment by thomasaarholt** [comment]
I think it's important to realise that the main purpose of the static website generators (sphinx, pelican, jekyll), is just to generate the structuring of the website - the backbone, so to speak. The speed and look of the website will be governed by the html and css code, which is partly done by the generator (via the Themes), and partly by whatever we want to do to make the front page shiny.

Sphinx will probably work fine, but having spent some time looking at Pelican and Jekyll, I prefer Jekyll, as a generator. When it comes to the look of the front page + main subpages (not documentation), that is best done by drawing out on A4 paper what we envisage. The jupyter page is probably a good starting point.

**Comment by thomasaarholt** [comment]
When it comes to the look of the front page, I too value speed over fanciness, and would prefer a "static page" (no movement in things as you scroll). It might be nice to show a few images of what using Hyperspy actually looks like - I'm sure between us we could dig up nice, highly zoomed-in screenshots of the qtconsole, notebook and HyperspyUI in action. Preferably with some sexy data processing taking place.

**Comment by francisco-dlp** [comment]
That's a good idea: currently we have highlights but they're hardly meaningful for anybody not already familiar with HyperSpy. Replacing them with e.g. a carrusel of pictures with captions would probably be a lot more effective.

**Comment by tjof2** [comment]
I agree with @to266 - Jupyter's layout is excellent. In particular, the navbar at the top stays visible throughout. The other example that springs to mind is Beaker, from the NY hedge fund TwoSigma: http://beakernotebook.com/

**Comment by francisco-dlp** [comment]
http://beakernotebook.com/ is really cool!

**Comment by thomasaarholt** [comment]
We're gonna need a pretty picture, lads. Any suggestions? Something that can replace the jupytercon NY conference picture. I'm currently modifying the Jupyter website, and have tried using screenshots from the jupyter notebook and UI, and it just doesn't look great.

Anyone got a nice high-res microscope image I can use? Happy to use anything else later (especially if francisco wants to push the non-EM aspect of hyperspy), but for now it would be nice to have something mildly related.

**Comment by thomasaarholt** [comment]
I've put together a draft/suggestion for a new Hyperspy website at https://thomasaarholt.github.io/. Please take a look. Obviously the picture will change (ideas welcome).

Plenty of things need work, but I decided not to invest more time in it right now.

**Comment by tjof2** [comment]
@thomasaarholt  a better URL would be https://thomasaarholt.github.io/ :-)

Change the navbar to a solid colour on loading (rather than the current behaviour) and I like it!

**Comment by thomasaarholt** [comment]
You are quite right :) Updated!

**Comment by vidartf** [comment]
I'd also pick another color than Jupyter orange :)
